LISISO/commonfunctions.sh

```diff
@@ -315,3 +315,54 @@ function checkrpms()
 		echo "RPM's are missing"
 	fi
 }
+
+function CheckPartitionSize()
+{
+        if (( $1 < $2 ));then
+                echo "LIS installation aborted due to insufficiant space"
+                printf "\t %s \t\t %s\n" "$3" "available: $1 required: $2 Bytes"
+                return 0
+        fi
+        return 1
+}
+
+function CheckRequiredSpace()
+{
+	No_of_symvers=$(($(ls /boot | grep "symvers" | wc -l)))
+	# 160MB(167772160 Bytes) minimum space is required for 1 kernel
+        MIN_SPACE_FOR_RAMFS_CREATION=167772160
+        lib_module_size_factor=1
+	# 250MB(268435456 Bytes) minimum space is required for more than 1 kernel
+	[ $No_of_symvers -gt 1 ] && MIN_SPACE_FOR_RAMFS_CREATION=268435456;lib_module_size_factor=2
+	lib_module_folder="/lib/modules"
+        boot_folder="/boot"
+	root_partition=$(df $lib_module_folder | grep -v Used | awk '{ print $1}')
+        boot_partition=$(df $boot_folder | grep -v Used | awk '{ print $1}')
+
+	ramdisk_size_factor=1
+	[ $root_partition != $boot_partition ] && ramdisk_size_factor=2
+
+        root_part_avail_space=$(echo $(($(stat -f --format="%a*%S" $lib_module_folder))))
+        boot_part_avail_space=$(echo $(($(stat -f --format="%a*%S" $boot_folder))))
+	
+	lib_module_required_space=$(rpm --queryformat='%{SIZE}' -qp kmod-microsoft-hyper*x86_64.rpm)
+        ramdisk_required_space=$(stat /boot/initramfs-$(uname -r).img --format="%s")
+
+        boot_part_required_space=$ramdisk_required_space
+        if [ $root_partition != $boot_partition ];then
+		root_part_required_space=`expr $MIN_SPACE_FOR_RAMFS_CREATION + $ramdisk_size_factor \* $ramdisk_required_space + $lib_module_size_factor \* $lib_module_required_space`
```

Comment:
Use $(...) notation instead of legacy backticks.   Here and everywhere.

also note expr is antiquated. Consider rewriting this using $((..)), ${} or [[ ]]